itools-packager/src/main/java/com/powsybl/itools/ItoolsPackagerMojo.java

```diff
@@ -79,33 +81,125 @@ public void setFiles(File[] files) {
-        try (ZipArchiveOutputStream zos = new ZipArchiveOutputStream(Files.newOutputStream(zipFilePath))) {
-            Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
-                @Override
-                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
-                    ZipArchiveEntry entry = new ZipArchiveEntry(baseDir.relativize(file).toString());
-                    if (Files.isExecutable(file)) {
-                        entry.setUnixMode(0100770);
-                    } else {
-                        entry.setUnixMode(0100660);
-                    }
-                    zos.putArchiveEntry(entry);
-                    Files.copy(file, zos);
-                    zos.closeArchiveEntry();
-                    return FileVisitResult.CONTINUE;
+    public void createPackage(Path dir, final Path baseDir, String fileName, String packageName) throws IOException {
+        Objects.requireNonNull(dir);
+        Objects.requireNonNull(baseDir);
+        Objects.requireNonNull(fileName);
+
+        String extension = compareFileExtension(fileName);
+        String newFileName = fileName;
+        // invalid extension
+        if (extension.equals("")) {
+            extension = ".zip";
+            newFileName = packageName + extension;
+        }
+
+        Path packageFilePath = baseDir.resolve(newFileName);
+        FileOutputStream fos = new FileOutputStream(String.valueOf(packageFilePath));
+
+        getLog().info("Generate package file : " + newFileName);
+
+        switch (extension) {
+            case ".zip":
+                try (ZipArchiveOutputStream zos = new ZipArchiveOutputStream(fos)) {
+                    Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+                        @Override
+                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                            ZipArchiveEntry entry = new ZipArchiveEntry(baseDir.relativize(file).toString());
+                            if (Files.isExecutable(file)) {
+                                entry.setUnixMode(0100770);
+                            } else {
+                                entry.setUnixMode(0100660);
+                            }
+                            zos.putArchiveEntry(entry);
+                            Files.copy(file, zos);
+                            zos.closeArchiveEntry();
+                            return FileVisitResult.CONTINUE;
+                        }
+
+                        @Override
+                        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                            zos.putArchiveEntry(new ZipArchiveEntry(baseDir.relativize(dir).toString() + "/"));
+                            zos.closeArchiveEntry();
+                            return FileVisitResult.CONTINUE;
+                        }
+                    });
+                }
```

Comment:
To reduce the complexity of this method, you should extract this visitor to a separate file.

Reply:
Ok